### PR TITLE
Added utility function outdims to compute output dimensions of a layer

### PR DESCRIPTION
**Pull request by darsnack**
Based on Slack chatter, I added a utility function, `outdims`, that computes the output dimensions for given input dimensions.

Example
```julia
layer = Conv((3, 3), 3 => 16)
outdims(layer, (10, 10)) # returns (8, 8)
```

### CODE REVIEW

File: test/layers/conv.jl [hunk 1]
```diff
@@ -107,3 +107,23 @@ end
     true
   end
 end
+
+@testset "conv output dimensions" begin
+  m = Conv((3, 3), 3 => 16)
```
Comment (baggepinnen) [inline test/layers/conv.jl]: Would be good to have some tests for different padding, stride and dilation

**Comment by MikeInnes** [comment]
I can see this being useful. Note that not every method needs a docstring; you can just give one docstring with a couple of examples, otherwise the docs for `outdims` will be pretty noisy.

You can probably reuse the tools in NNlib to calculate this for convolutions, which will make it much less likely to miss anything.

**Comment by darsnack** [comment]
Okay switched to NNlib.jl, added some more tests w.r.t. pad, stride, etc., and also added a small section to the docs.

**Comment by darsnack** [comment]
Any ideas on the build errors? I don't really follow the stack trace, and I am not able to reproduce it on my local system.

**Comment by MikeInnes** [comment]
Quite unlikely to be your fault, but we should get it passing. If you push a new commit the error might just go away.

**Comment by darsnack** [comment]
> Quite unlikely to be your fault, but we should get it passing. If you push a new commit the error might just go away.

So I guess this didn't work. Have you seen this issue pop up with any other builds since I last pushed?

**Comment by MikeInnes** [comment]
We have had a bit of flakiness lately. 

bors try

**Comment by ianshmean** [comment]
It would be great to revive this idea. Perhaps worth rebasing and re-running tests?

**Comment by darsnack** [comment]
@MikeInnes looks like the build is passing now? Though Travis is only testing for Julia 1.3?

**Comment by dhairyagandhi96** [comment]
bors try

Let's try this once to see if things are settled down. Post which thoughts on documenting it's scope, since it might be odd to get it running with arbitrary functions?

**Comment by darsnack** [comment]
Sure I can push a commit updating the doc section to specify which layers it works on?

**Comment by ianshmean** [comment]
Perhaps the docs could state which layers it works for out of the box, and give the outdims conv method code to give a handy starting point for people to generate an outdims method for custom layers?

**Comment by ianshmean** [comment]
This is looking good to me and I'm keen to start using these functions. 
Is it ready to be merged?

**Comment by darsnack** [comment]
> Is it ready to be merged?

I am satisfied with merging it in the current form.

**Comment by dhairyagandhi96** [comment]
Happy to merge this now
bors r+

**Pull request merged**